src/Compilers/CSharp/Portable/Symbols/OverriddenOrHiddenMembersResult.cs

```diff
@@ -52,6 +53,37 @@ internal sealed class OverriddenOrHiddenMembersResult
             }
         }
 
+        internal static Symbol GetOverriddenMember(Symbol substitutedOverridingMember, Symbol overriddenByDefinitionMember)
+        {
+            Debug.Assert(!substitutedOverridingMember.IsDefinition);
+
+            if ((object)overriddenByDefinitionMember != null)
+            {
+                NamedTypeSymbol overridenByDefinitionContainingTypeDefinition = overriddenByDefinitionMember.ContainingType.OriginalDefinition;
+                for (NamedTypeSymbol baseType = substitutedOverridingMember.ContainingType.BaseTypeNoUseSiteDiagnostics; 
+                    (object)baseType != null; 
+                    baseType = baseType.BaseTypeNoUseSiteDiagnostics)
+                {
+                    if (baseType.OriginalDefinition == overridenByDefinitionContainingTypeDefinition)
+                    {
+                        if (baseType.IsDefinition)
+                        {
+                            return overriddenByDefinitionMember.OriginalDefinition;
+                        }
+
+                        if (baseType == overriddenByDefinitionMember.ContainingType)
```

Comment:
Also, consider adding a comment that this is just an optimization of the expression below (which would also be correct).